### PR TITLE
fix: path on windows

### LINKED ISSUE #88: GetDefinitionForFilename fails for windows paths
**Platform:**
Windows

**Version:**
Tested with current master

**Testcase** 
`.editorconfig`:
```
root = true
[*]
indent_style = tab

[*.ini]
indent_style = space

[test.toml]
indent_style = space
```

`test.ini`
```
[foo]
  bar = "buz"
```

`test.toml`
```
[foo]
  bar = "buz"
```

**Expected behavior**
I expected to get `IndentStyleSpaces` for `test.toml` and `test.ini` with windows paths.
but only `test.ini` returns the expected indent style.

**Unit Test**
```
func TestGetDefinition(t *testing.T) {
       ...

	def, err = ec.GetDefinitionForFilename("C:\\home\\test.toml")
	if err != nil {
		t.Errorf("Couldn't parse file: %v", err)
	}
	assert.Equal(t, IndentStyleSpaces, def.IndentStyle)

	def, err = ec.GetDefinitionForFilename("C:\\home\\test.ini")
	if err != nil {
		t.Errorf("Couldn't parse file: %v", err)
	}
	assert.Equal(t, IndentStyleSpaces, def.IndentStyle)
```

**Problem**
The generated Regular Expression `^(?:/|/.*/)test\.toml$` does not match the path `/C:\\Users\\tom-tomsen\\test.toml`

### PR DESCRIPTION
**Pull request by greut**
Closes #88

### CODE REVIEW

**Comment by greut** [comment]
:point_down: a build for windows.
[editorconfig-core-go_v2.3.8-development_windows_amd64.zip](https://github.com/editorconfig/editorconfig-core-go/files/5577880/editorconfig-core-go_v2.3.8-development_windows_amd64.zip)

**Comment by mstruebing** [comment]
@tomtomsen

File: editorconfig.go [hunk 1]
```diff
@@ -122,7 +123,11 @@ func (e *Editorconfig) GetDefinitionForFilename(name string) (*Definition, error
 		}
 
 		if !strings.HasPrefix(name, "/") {
```
Comment (greut) [inline editorconfig.go:125]: this will probably break if you give a full path instead of a relative one, e.g. `C:\Users\me\foo.toml`

**Comment by greut** [comment]
Tthe way to use this is.

```console
> editorconfig.exe test.toml
indent_style = space

```

**Pull request merged**